relay-general/src/store/transactions/processor.rs

```diff
@@ -342,26 +348,30 @@ impl Processor for TransactionsProcessor<'_> {
         }
 
         if self.name_config.scrub_identifiers {
-            // Apply the rule if any found
-            self.apply_transaction_rename_rule(
-                &mut event.transaction,
-                event.transaction_info.value_mut(),
-            )?;
-
             // Normalize transaction names for URLs and Sanitized transaction sources.
             // This in addition to renaming rules can catch some high cardinality parts.
             if matches!(
                 event.get_transaction_source(),
                 &TransactionSource::Url | &TransactionSource::Sanitized
             ) {
                 scrub_identifiers(&mut event.transaction)?;
-                if self.name_config.mark_scrubbed_as_sanitized {
-                    event
-                        .transaction_info
-                        .get_or_insert_with(Default::default)
-                        .source
-                        .set_value(Some(TransactionSource::Sanitized));
-                }
+            }
+
+            self.apply_transaction_rename_rule(
+                &mut event.transaction,
+                event.transaction_info.value_mut(),
+            )?;
+
+            if !matches!(
+                event.get_transaction_source(),
+                &TransactionSource::Sanitized
```

Comment:
Wouldn't this set `source=sanitized` for every source (`route`, `custom`, etc.). I think we need to match on `TransactionSource::Url` here explicitly.

Reply:
Heh, I think I got confused _again_ from the code structure. Let's see when we can remove the feature flags here and make the code a bit easier to follow.

Thanks for putting in the effort!